tests/integration/seed/seedsVerification.test.js

```diff
@@ -656,16 +665,158 @@ describe('SEEDS VERIFICATION', () => {
+              TRAINEE_DELETION,
+            ];
+
+            const hasRofOrVendorAdminRole = [TRAINING_ORGANISATION_MANAGER, VENDOR_ADMIN]
+              .includes(get(ch.createdBy, 'role.vendor.name'));
+            const hasCourseEditionRole = [TRAINING_ORGANISATION_MANAGER, VENDOR_ADMIN, TRAINER]
+              .includes(get(ch.createdBy, 'role.vendor.name')) ||
+              [COACH, CLIENT_ADMIN].includes(get(ch.createdBy, 'role.client.name'));
+
+            if (rofOrVendorAdminActions.includes(ch.action) && !hasRofOrVendorAdminRole) return false;
+            if (otherActions.includes(ch.action) && !hasCourseEditionRole) return false;
+
+            return true;
+          });
+          expect(everyHistoryCreatorHasGoodRole).toBeTruthy();
+        });
+
+        it('should pass if user who created history is allowed to access course', () => {
+          const everyHistoryCreatorIsAllowedToAccessCourse = courseHistoryList.every((ch) => {
+            const hasRofOrVendorAdminRole = [TRAINING_ORGANISATION_MANAGER, VENDOR_ADMIN]
+              .includes(get(ch.createdBy, 'role.vendor.name'));
+
+            const isCourseTrainer = [TRAINER].includes(get(ch.createdBy, 'role.vendor.name')) &&
+              UtilsHelper.areObjectIdsEquals(ch.createdBy._id, ch.course.trainer);
+
+            const isFromCompanyLinkedToCourse = [COACH, CLIENT_ADMIN].includes(get(ch.createdBy, 'role.client.name')) &&
+              ch.createdBy.userCompanyList.some(uc => UtilsHelper.doesArrayIncludeId(ch.course.companies, uc.company));
+
+            if (hasRofOrVendorAdminRole || isCourseTrainer || isFromCompanyLinkedToCourse) return true;
+
+            return false;
+          });
+          expect(everyHistoryCreatorIsAllowedToAccessCourse).toBeTruthy();
+        });
+
+        it('should pass if trainee is in course for trainee_addition action', () => {
+          const isEveryTraineeInCourse = courseHistoryList
+            .filter(ch => [TRAINEE_ADDITION].includes(ch.action))
+            .every((ch) => {
+              if (UtilsHelper.doesArrayIncludeId(ch.course.trainees, ch.trainee._id)) return true;
+
+              const traineeDeletions = courseHistoryList
+                .filter(
+                  history => history.action === TRAINEE_DELETION &&
+                  UtilsHelper.areObjectIdsEquals(history.trainee._id, ch.trainee._id) &&
+                  UtilsHelper.areObjectIdsEquals(history.course._id, ch.course._id)
+                );
+
+              const traineeAdditions = courseHistoryList
+                .filter(
+                  history => history.action === TRAINEE_ADDITION &&
+                  UtilsHelper.areObjectIdsEquals(history.trainee._id, ch.trainee._id) &&
+                  UtilsHelper.areObjectIdsEquals(history.course._id, ch.course._id)
+                );
+
+              return traineeDeletions.length === traineeAdditions.length;
+            });
+
+          expect(isEveryTraineeInCourse).toBeTruthy();
```

Comment:
Est-ce que ce ne serait pas un doublon avec le test "should pass if all trainees are in course company at registration', " ?